File: examples-sdk/python/406-agent-vil-handler-shm.py
#!/usr/bin/env python3
"""406-agent-vil-handler-shm — Python SDK equivalent
Compile: vil compile --from python --input 406-agent-vil-handler-shm.py --release
"""
import os
from vil import VilPipeline, VilServer, ServiceProcess

server = VilServer("fraud-detection-agent", port=3126)
fraud_agent = server.service_process("fraud-agent")
fraud_agent.endpoint("POST", "/detect", "detect_fraud")
fraud_agent.endpoint("GET", "/health", "health")
server.compile()

File: examples-sdk/typescript/406-agent-vil-handler-shm.ts
#!/usr/bin/env tsx
// 406-agent-vil-handler-shm — TypeScript SDK equivalent
// Compile: vil compile --from typescript --input 406-agent-vil-handler-shm.ts --release

import { VilPipeline, VilServer, ServiceProcess } from "vil-sdk";

const server = new VilServer("fraud-detection-agent", 3126);
const fraud_agent = new ServiceProcess("fraud-agent");
fraud_agent.endpoint("POST", "/detect", "detect_fraud");
fraud_agent.endpoint("GET", "/health", "health");
server.service(fraud_agent);
server.compile();
